### PR TITLE
Creating unit tests

### PR DESCRIPTION
**Pull request by keelimeguy**

### CODE REVIEW

File: sentiment_analysis_tools/sentiments.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    def sentiment_analysis_flair(text: str) -> float:
+        """
+        Run sentiment analysis using the library flair. Returns default sentiment
+
+        Works based on a character-level LSTM neural network
+        :param text: text to be analyzed
+        :return: sentiment for given text
+        """
+        if not text:
+            return 0
```
Comment (v2thegreat) [inline sentiment_analysis_tools/sentiments.py:82]: Use NaN instead of 0 for this

File: sentiment_analysis_tools/sentiments.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    def sentiment_analysis_textblob(text: str) -> float:
+        """
+        Run sentiment analysis using the library textblob. Returns default sentiment
+
+        Works similar to NLTK's sentiment analysis, but includes subjectivity analysis
+        :param text: text to be analyzed
+        :return: sentiment for given text
+        """
+        if not text:
+            return 0
```
Comment (v2thegreat) [inline sentiment_analysis_tools/sentiments.py:68]: Use NaN instead of 0 for this

File: Dozent/tests/test_downloader_tools.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
+import unittest
```
Comment (v2thegreat) [inline Dozent/tests/test_downloader_tools.py:1]: Move `tests` to it's own directory

**Review by v2thegreat — CHANGES_REQUESTED** [review]

File: .github/workflows/python-app.yml [hunk 4]
```diff
@@ -3,11 +3,7 @@
 
 name: Build
 
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+on: [push, pull_request, workflow_dispatch]
```
Comment (v2thegreat) [inline .github/workflows/python-app.yml:6]: Any major reason for switching this? Wouldn't it be better to only run this only for the master branch and run the tests manually? Just want your thoughts

File: sentiment_analysis_tools/sentiments.py [hunk 5]
```diff
@@ -61,6 +65,9 @@ def sentiment_analysis_textblob(text: str) -> float:
         :param text: text to be analyzed
         :return: sentiment for given text
         """
+        if not text:
+            return math.nan
```
Comment (v2thegreat) [inline sentiment_analysis_tools/sentiments.py:69]: Good add

File: tests/test_sentiments.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+class SentimentsTestCase(unittest.TestCase):
+    def test_multiple_sentiment_analysis_empty(self):
+        expected_output = {
+            'nltk': math.nan,
+            'textblob': math.nan,
+            'flair': math.nan
+        }
+
+        self.assertEqual(expected_output, Sentiments.multiple_sentiment_analysis(''))
```
Comment (v2thegreat) [inline tests/test_sentiments.py:15]: Good test. We might want to add something for future tests to check if common sentences are positive or negative

**Review by v2thegreat — APPROVED** [review]

File: .github/workflows/python-app.yml (hunk 4 above)
Comment (keelimeguy) [reply .github/workflows/python-app.yml:6]: I don't have access to the API for this branch (missing permissions) and the button does not show up in the GUI except for on actions of the master branch. Therefore, since these PRs have been going to Dozent, there has been no way for me to manually trigger a test.

Comment (keelimeguy) [reply .github/workflows/python-app.yml:6]: Not to mention, I have never seen the GUI button show up for the manual workflow_dispatch hook, so I have doubts if it even works. I've only been able to trigger manual tests in my own repos using the API

Comment (keelimeguy) [reply .github/workflows/python-app.yml:6]: I'll correct myself, I can see the button on my forked repo, but not here. Likely a permissions issue still.

image
